chia/rpc/wallet_rpc_api.py

```diff
@@ -264,15 +264,15 @@ async def get_latest_coin_spend(
                 coin_state_list = await self.service.wallet_state_manager.wallet_node.fetch_children(
                     coin_state.coin.name(), peer=peer
                 )
-                odd_coin = 0
+                odd_coin = None
                 for coin in coin_state_list:
                     if coin.coin.amount % 2 == 1:
-                        odd_coin += 1
-                    if odd_coin > 1:
-                        raise ValueError("This is not a singleton, multiple children coins found.")
-                if odd_coin == 0:
+                        if odd_coin is not None:
+                            raise ValueError("This is not a singleton, multiple children coins found.")
+                        odd_coin = coin
+                if odd_coin is None:
                     raise ValueError("Cannot find child coin, please wait then retry.")
-                coin_state = coin_state_list[0]
+                coin_state = odd_coin
```

Comment:
my reading of this is that the previous code, incorrectly, assumed that the odd coin was always the first one. This fixes it by both ensuring there's only one and returning the correct coin.

is that right?